File: services/networkx_service/src/process/algorithm/coloring/dsatur.py
import networkx as nx
from algorithm.algorithm import Algorithm


class DSatur(Algorithm):
    def __init__(self, instance, publishers):
        super().__init__(instance, publishers)

    def algorithm(self):
        color_map = nx.coloring.greedy_color(self.graph, strategy="DSATUR")
        num_colors = max(color_map.values()) + 1

        return num_colors

File: services/graph4j_service/src/process/src/main/java/process/algorithm/coloring/DSatur.java
package process.algorithm.coloring;

import org.graph4j.alg.coloring.DSaturGreedyColoring;
import process.algorithm.Algorithm;
import process.instance.Instance;
import process.pubsub.Publisher;

import java.util.List;

public class DSatur extends Algorithm {

    public DSatur(Instance instance, List<Publisher> publishers) {
        super(instance, publishers);
    }

    public void algorithm() {
        DSaturGreedyColoring dSaturGreedyColoring = new DSaturGreedyColoring(this.graph);

        bestResult = dSaturGreedyColoring.findColoring().numUsedColors();
    }
}
